### PR TITLE
feat: Add supports for vDPA interface

### PR DESCRIPTION
**Pull request by makoto126**
Tested with [SR-IOV Network Device Plugin](https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin) and modified kube-ovn(ovs-dpdk offload to our DPU).
Live migration previously tested in Kubevirt(with qemu).

**VM spec**
```yaml
apiVersion: virt.virtink.smartx.com/v1alpha1
kind: VirtualMachine
spec:
  instance:
    interfaces:
      - name: pod
      - name: vdpa1
        vdpa: { numQueues: 9, iommu: False }
      - name: vdpa2
        vdpa: { numQueues: 9, iommu: False }
  networks:
    - name: pod
      pod: {}
    - name: vdpa1
      multus:
        networkName: offload-ovn1
    - name: vdpa2
      multus:
        networkName: offload-ovn2
```
**SR-IOV device plugin config**
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: sriovdp-config
  namespace: kube-system
data:
  config.json: |
    {
        "resourceList": [
            {
                "resourceName": "jaguar",
                "resourcePrefix": "jaguarmicro.com",
                "selectors": {
                    "vendors": ["1f53"],
                    "devices": ["1000"],
                    "drivers": ["jaguar"],
                    "vdpaType": "vhost"
                }
            }
        ]
    }
```
**NetworkAttachmentDefinition Spec**
```
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: offload-ovn1
  namespace: default
  annotations:
    k8s.v1.cni.cncf.io/resourceName: jaguarmicro.com/jaguar
spec:
  config: >-
    {
        "cniVersion": "0.3.0",
        "type": "kube-ovn",
        "server_socket": "self.dpu.com:12345",
        "provider": "offload-ovn1.default.ovn"
    }
---
apiVersion: k8s.cni.cncf.io/v1
kind: NetworkAttachmentDefinition
metadata:
  name: offload-ovn2
  namespace: default
  annotations:
    k8s.v1.cni.cncf.io/resourceName: jaguarmicro.com/jaguar
spec:
  config: >-
    {
        "cniVersion": "0.3.0",
        "type": "kube-ovn",
        "server_socket": "self.dpu.com:12345",
        "provider": "offload-ovn2.default.ovn"
    }
```
**Pod network status**
```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    k8s.v1.cni.cncf.io/network-status: |-
      [{
          "name": "cbr0",
          "interface": "eth0",
          "ips": [
              "10.244.2.45"
          ],
          "mac": "5a:61:29:39:0d:29",
          "default": true,
          "dns": {},
          "gateway": [
              "10.244.2.1"
          ]
      },{
          "name": "default/offload-ovn1",
          "interface": "net1",
          "ips": [
              "10.1.0.20"
          ],
          "mac": "00:00:00:03:6C:84",
          "dns": {},
          "device-info": {
              "type": "vdpa",
              "version": "1.1.0",
              "vdpa": {
                  "parent-device": "test1",
                  "driver": "vhost",
                  "path": "/dev/vhost-vdpa-2",
                  "pci-address": "0000:68:00.5"
              }
          }
      },{
          "name": "default/offload-ovn2",
          "interface": "net2",
          "ips": [
              "10.2.0.2"
          ],
          "mac": "00:00:00:50:A4:81",
          "dns": {},
          "device-info": {
              "type": "vdpa",
              "version": "1.1.0",
              "vdpa": {
                  "parent-device": "test2",
                  "driver": "vhost",
                  "path": "/dev/vhost-vdpa-3",
                  "pci-address": "0000:68:00.6"
              }
          }
      }]
```
**VM after dhclient**
```bash
ubuntu@ubuntu-container-rootfs:~$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: ens4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 5a:61:29:39:0d:29 brd ff:ff:ff:ff:ff:ff
    altname enp0s4
    inet 10.244.2.45/24 metric 100 brd 10.244.2.255 scope global ens4
       valid_lft forever preferred_lft forever
3: ens6: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:00:00:03:6c:84 brd ff:ff:ff:ff:ff:ff
    altname enp0s6
    inet 10.1.0.20/24 brd 10.1.0.255 scope global dynamic ens6
       valid_lft 2305sec preferred_lft 2305sec
4: ens7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:00:00:50:a4:81 brd ff:ff:ff:ff:ff:ff
    altname enp0s7
    inet 10.2.0.2/24 brd 10.2.0.255 scope global dynamic ens7
       valid_lft 2305sec preferred_lft 2305sec
```

### CODE REVIEW

File: pkg/apis/virt/v1alpha1/types.go [hunk 1]
```diff
@@ -115,6 +116,11 @@ type InterfaceMasquerade struct {
 type InterfaceSRIOV struct {
 }
 
+type InterfaceVDPA struct {
+	NumQueues int  `json:"numQueues,omitempty"`
+	Iommu     bool `json:"iommu,omitempty"`
```
Comment (fengye87) [inline pkg/apis/virt/v1alpha1/types.go:121]: Better name it IOMMU to keep it in Golang style

**Review by fengye87 — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please also doc this new feature in [this file](https://github.com/smartxworks/virtink/blob/main/docs/interfaces_and_networks.md)

File: pkg/apis/virt/v1alpha1/types.go (hunk 1 above)
Comment (makoto126) [reply pkg/apis/virt/v1alpha1/types.go:121]: The VdpaConfig struct and other structs in pkg/cloudhypervisor/client.go used Iommu too.
```golang
type VdpaConfig struct {
	Id         string `json:"id,omitempty"`
	Iommu      bool   `json:"iommu,omitempty"`
	NumQueues  int    `json:"num_queues"`
	Path       string `json:"path"`
	PciSegment int16  `json:"pci_segment,omitempty"`
}
```

Comment (fengye87) [reply pkg/apis/virt/v1alpha1/types.go:121]: Yes, that's generated. For non-generated code, better keep them in style

Comment (makoto126) [reply pkg/apis/virt/v1alpha1/types.go]: @fengye87 I updated the doc and IOMMU.

**Review by fengye87 — APPROVED** [review]

**Pull request merged**